lib/game_client/ogs/ogs_websocket_manager.dart

```diff
@@ -0,0 +1,245 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:uuid/uuid.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+import 'package:web_socket_channel/status.dart' as status;
+
+class OGSWebSocketManager {
+  WebSocketChannel? _channel;
+  final String serverUrl;
+  String? _deviceId;
+  
+  final ValueNotifier<bool> _connected = ValueNotifier(false);
+  final StreamController<Map<String, dynamic>> _messageController = 
+      StreamController<Map<String, dynamic>>.broadcast();
+  
+  Timer? _pingTimer;
+  Timer? _reconnectTimer;
+  int _reconnectAttempts = 0;
+  static const int maxReconnectAttempts = 10;
+  
+  // For latency tracking
+  double _latency = 0.0;
+  double _clockDrift = 0.0;
+  
+  OGSWebSocketManager({required this.serverUrl}) {
+    _generateDeviceId();
+  }
+  
+  ValueNotifier<bool> get connected => _connected;
+  Stream<Map<String, dynamic>> get messages => _messageController.stream;
+  
+  void _generateDeviceId() {
+    // Generate a proper UUID for device ID like OGS frontend
+    const uuid = Uuid();
+    _deviceId = uuid.v4();
+  }
+  
+  Future<void> connect() async {
+    if (_channel != null) {
+      await disconnect();
+    }
+    
+    try {
+      // Convert HTTP(S) URL to WebSocket URL
+      final wsUrl = serverUrl
+          .replaceFirst('https://', 'wss://')
+          .replaceFirst('http://', 'ws://');
+      
+      debugPrint('Connecting to OGS WebSocket: $wsUrl');
```

Comment:
I'm using `Logger` for this purpose. See example usage here: https://github.com/ale64bit/WeiqiHub/blob/main/lib/play/ai_game_page.dart#L84. This makes it easier in the debug console and to control it globally at the beginning of the app.

Reply:
Oh great! I like log levels.

One think I noticed: I run the app from the command line, but don't see logs in terminal.  Do they just show up if I run via VS Code/Debug console?